main.go

```diff
@@ -61,26 +61,28 @@ func udpServer(address string, h *hub) {
 	for {
 		n, addr, err := serverConn.ReadFromUDP(buf)
 		if err != nil {
-			fmt.Println("Error: UDP read: %v", err)
+            log.WithFields(log.Fields{
+                "error": err,
+            }).Fatal("UDP read error")
 		}
 		if strings.HasPrefix(string(buf[:n]), "CONNECT") {
 			split := strings.Split(string(buf[:n]), ":")
```

Comment:
Again, use a struct or helper functions for regex parsing and accessing elements in `buf`